triqs/operators/many_body_operator.hpp

```diff
@@ -379,6 +380,16 @@ namespace operators {
   }
  };
 
+ // Assert that two many-body-operators are term-wise close
+ template<typename scalar1_t, typename scalar2_t>
+ void assert_operators_are_close(many_body_operator_generic<scalar1_t> const &op1, many_body_operator_generic<scalar2_t> const &op2, double precision){ 
+  auto terms_are_equal = [precision]( auto const &t1, auto const &t2 ){ using std::abs; return (t1.monomial == t2.monomial) && abs(t1.coef - t2.coef) < precision; }; 
```

Comment:
For the comparison, I would stick to the pattern used in the rest of the class:
```c++
auto terms_are_equal = [precision]( auto const &t1, auto const &t2 ){
  using triqs::utility::is_zero;
  return (t1.monomial == t2.monomial) && is_zero(t1.coef - t2.coef, precision);
};
```

Unfortunately, my implementation of `is_zero()` was a bit sloppy, so I also suggest updating `/triqs/utility/numeric_ops.hpp`:
```c++
//
// Zero value tests
//
template<typename T> // Integral types
std14::enable_if_t<std::is_integral<T>::value,bool> is_zero(T const& x, T = 0) {
 return x==0;
}

template<typename T> // Floating-point types
std14::enable_if_t<std::is_floating_point<T>::value,bool> is_zero(T const& x, T tolerance = 100*std::numeric_limits<T>::epsilon()) {
 return std::abs(x) < tolerance;
}

template<typename T> // std::complex
std14::enable_if_t<triqs::is_complex<T>::value,bool> is_zero(T const& x, T tolerance = 100*std::numeric_limits<typename T::value_type>::epsilon()) {
 return is_zero(std::real(x), std::real(tolerance)) && is_zero(std::imag(x), std::real(tolerance));
}
```